encoders/numeric/TruncatedSVDEncoder/__init__.py

```diff
@@ -0,0 +1,66 @@
+__copyright__ = "Copyright (c) 2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+from jina.executors.encoders.numeric import TransformEncoder
+from jina.executors.decorators import batching, require_train
+
+
+class TruncatedSVDEncoder(TransformEncoder):
+    """
+    Encodes data using truncated SVD, and does not center the data before
+    computing SVD hence making them efficient for sparse input matrices.
+
+    Encodes data from a ndarray of size `B x T` into a ndarray of size `B x D`.
+    Where `B` is the batch's size and `T` and `D` are the dimensions pre (`T`)
+    and after (`D`) the encoding.
+
+    :param output_dim: Dimension of the output embedded space
+    :param algorithm: algorithm to be used to run SVD
+    :param max_iter: Number of iterations for the encoder to run
+    :param args:  Additional positional arguments
+    :param kwargs: Additional keyword arguments
+
+    More details can be found
+    `here <https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.TruncatedSVD.html>`_
+
+    .. note::
+        :class:`TruncatedSVDEncoder` must be trained before calling ``encode()``.
+    """
+
+    def __init__(
+        self,
+        output_dim: int = None,
```

Comment:
We should have a better default value, None I guess will not be supported by the backend

Reply:
Good catch, sure, I will update that, thanks.